data/lib/src/datasource_impl/document_datasource_impl.dart

```diff
@@ -127,4 +127,60 @@ class DocumentDataSourceImpl implements DocumentDataSource {
 
     return DownloadTaskId(taskId);
   }
+
+  @override
+  Stream<Uri> downloadDocumentIOS(Document document, Token token, Uri baseUrl, CancelToken cancelToken) {
+    final streamController = StreamController<Uri>();
+
+    try {
+      Future.wait([
+        _linShareHttpClient.downloadDocumentIOS(Endpoint.documents
+            .downloadServicePath(document.documentId.uuid)
+            .generateDownloadUrl(baseUrl), cancelToken, token),
+        getTemporaryDirectory()
+      ]).then((values) {
+        final fileStream = values[0] as Stream<Uint8List>;
+        final tempFilePath = '${(values[1] as Directory).absolute.path}/${document.name}';
+
+        final file = File(tempFilePath);
+        file.createSync(recursive: true);
+        var randomAccessFile = file.openSync(mode: FileMode.write);
+        StreamSubscription subscription;
+
+        subscription = fileStream.listen((data) {
```

Comment:
Have you tried something like `fileStream.takeWhile(_ => cancelToken == null || ... )` ? to avoid pause and resume

Reply:
cool, I can use `fileStream.takeWhile`. but for more information, `pause` and `resume` is for writing to file, you have to `pause` to write file then `resume` to write file again. So I gonna refactor it like this: 
```subscription = fileStream
            .takeWhile((_) => cancelToken == null || !cancelToken.isCancelled)
            .listen((data) {
          subscription.pause();
          randomAccessFile.writeFrom(data).then((_randomAccessFile) {
            randomAccessFile = _randomAccessFile;
            subscription.resume();
          })```